.github/workflows/build_docs.yml

```diff
@@ -0,0 +1,39 @@
+name: Build and commit docs
+
+on:
+  push:
+    branches: [ main ]
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  build:
+
+    # prevent subsequent commits from trigger the job multiple times
+    concurrency:
+      group: ci-${{ github.ref }}
+      cancel-in-progress: true
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup NodeJS
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16.x
+        cache: 'npm'
+    - name: Install and Build Docs
+      run: |
+        npm ci
+
+        sudo apt-get install hugo
+        hugo version
+        npm run build:docs -- --yes
```

Comment:
Can you break these up into each step and I think we can drop the version output? it should show up in the apt install bit.